src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AsyncLazy`1.cs

```diff
@@ -34,21 +29,21 @@ public static AsyncLazy<T> Create<T>(T value)
 /// cached for future requests or not. Choosing to not cache means the computation functions are kept
 /// alive, whereas caching means the value (but not functions) are kept alive once complete.
 /// </summary>
-internal sealed class AsyncLazy<T>
+internal class AsyncLazy<T, TData> : AsyncLazy<T>
```

Comment:
can you seal this?

Reply:
i would be 100% fine with this being a *private* nested type of `AsyncLazy<T>`.  `AsyncLazy<T>` could also have a *private* constructor.